### DISCUSSION TITLE
[Help] Using line in on HiFiBerry DAC+ ADC Pro headless Raspberry Pi OS - audio pipe appears to stop completely.

### DISCUSSION THREAD

**Discussion by Syco54645** [snapcast/snapcast#993] [Q&A] [answered]
So it seems that the documentation on how to use line-in with snapcast is severely lacking. Is there documentation anywhere that shows how to accomplish this? Info appears sparse though I have found a few instances of it being discussed but ultimately it appears that none of the discussions have not really provided a solution. I did find the pull request with documentation for using Sox but as written I am unable to get it working correctly. The issues could be related to the headless setup. Going to keep hammering it but figured I would make a post in Q&A.

Edit I see that I never stated the true issue I am having. The player (in this case my phone using a chromecast audio) will be 30 seconds ahead the actual audio that is playing. At some point (appears random) the pipe will stop and then the music starts over, which makes sense because we are piping to a file, pipe stops and the server just loops back around. Oddly I am not seeing the pipe stop in terminal. I have tried using parec, arecord, and sox. I assume something can be done with pulse audio that would hopefully be more reliable than a pipe. Going to keep investigating.

**Comment by Syco54645** [comment, marked as answer]
OK so I have found the answer to this on my own. I wrote a little guide to document what I did and created a PR for the project. Hopefully others will benefit from this. https://github.com/badaix/snapcast/pull/1002

**Comment by billiumbuck** [comment]
Hello Sysco54645
I followed your little guide but cannot make it work.
I get  

> Server::start: Can't open device 'hw:ALC892 Analog', error: No such device

 no matter what text I tried for device including number hw:1,1.  The line I put in snapserver.conf is
`stream = alsa:///?name=linein&device=hw:ALC892 Analog`

the output from arecord -l
`**** List of CAPTURE Hardware Devices ****
card 1: SB [HDA ATI SB], device 0: ALC892 Analog [ALC892 Analog]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: SB [HDA ATI SB], device 2: ALC892 Alt Analog [ALC892 Alt Analog]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
`

Could you (or any one who has this working) please advise what (and where) to put in snapserver.conf

Snapcast works great for me with MPD and some clients including android for which I am very grateful to all involved. The server runs on a headless Debian bookworm.

Thanks in anticipation.
